### ISSUE TITLE
Tests Failing Hard during Actions

### ISSUE THREAD

**Issue by confused-Techie** [confused-Techie/atom-community-server-backend-JS#17]
Looking at the last GH Test action run, its status is: `The job running on runner Hosted Agent has exceeded the maximum execution time of 360 minutes.`

It seems that the engine match has no failed when no engine is provided. Further investigate is needed into why this happens. The last logs available during the test are as follows:

```
● GET /api/packages/:packageName › Valid package, gives correct object
    TypeError: engine.match is not a function
      240 |   let reg =
      241 |     /(^\W*)([0-9]*).([0-9]*).([0-9]*)\s(\W*)([0-9]*).([0-9]*).([0-9]*)$/;
    > 242 |   let raw_engine = engine.match(/^([0-9]*).([0-9]*).([0-9]*)/);
          |                           ^
      243 |   let engine_semver = {
      244 |     major: raw_engine[1],
      245 |     minor: raw_engine[2],
      at Object.match [as EngineFilter] (src/collection.js:242:27)
      at Object.EngineFilter [as GETPackagesDetails] (src/handlers/package_handler.js:241:31)
      at src/main.js:219:3
  ● GET /api/packages/:packageName › Valid package, gives correct object
    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
      28 |
      29 | describe("GET /api/packages/:packageName", () => {
    > 30 |   test("Valid package, gives correct object", async () => {
         |   ^
      31 |     const res = await request(app).get("/api/packages/what-a-package");
      32 |     expect(res.body.name).toBe("what-a-package");
      33 |   });
      at test (src/tests/main.test.js:30:3)
      at Object.describe (src/tests/main.test.js:29:1)
```

**Comment by Digitalone1** [comment]
I didn't look at this. The merge request #18 may fix it. I worked on it because that regex was marked as an issue by Codacy.

**Comment by confused-Techie** [comment]
As suggested, resolved by @Digitalone1 pull with commit 7e19762878331d219dbbcc94bf045e311e212311
